Paxifax.cs

```diff
@@ -21,13 +21,186 @@
+        device.SendPacket(packet);
+#if DEBUG
+        Debug.WriteLine(PaxConfig.deviceMap[out_port].Name);
+      } else {
+        Debug.WriteLine("<dropped>");
+#endif
+      }
+    }
+  }
+
+  // Simple packet processor that can forward to multiple interfaces. It is "simple" because
+  // it can only transform the given packet, and cannot generate new ones.
+  public abstract class MultiInterface_SimplePacketProcessor : PacketProcessor {
+    // Return the offsets of network interfaces that "packet" is to be forwarded to.
+    abstract public ForwardingDecision process_packet (int in_port, ref Packet packet);
+
+    public static int[] broadcast (int in_port)
+    {
+      int[] out_ports = new int[PaxConfig.no_interfaces - 1];
+      // We retrieve number of interfaces in use from PaxConfig.
+      // Naturally, we exclude in_port from the interfaces we're forwarding to since this is a broadcast.
+      int idx = 0;
+      for (int ofs = 0; ofs < PaxConfig.no_interfaces; ofs++)
+      {
+        if (ofs != in_port)
+        {
+          out_ports[idx] = ofs;
+          idx++;
+        }
+      }
+      return out_ports;
+    }
+
+    public void packetHandler (object sender, CaptureEventArgs e)
+    {
+      var packet = PacketDotNet.Packet.ParsePacket(e.Packet.LinkLayerType, e.Packet.Data);
+      int in_port = PaxConfig.rdeviceMap[e.Device.Name];
+
+      int[] out_ports;
+      ForwardingDecision des = process_packet (in_port, ref packet);
+      if (des is ForwardingDecision.MultiPortForward)
+      {
+        out_ports = ((ForwardingDecision.MultiPortForward)des).target_ports;
+      } else {
+        throw (new Exception ("Expected SinglePortForward"));
+      }
+
+#if DEBUG
+      Debug.Write(PaxConfig.deviceMap[in_port].Name + " -> ");
+#endif
+      for (int idx = 0; idx < out_ports.Length; idx++)
+      {
+        PaxConfig.deviceMap[out_ports[idx]].SendPacket(packet);
+#if DEBUG
+        if (idx < out_ports.Length - 1)
+        {
+          Debug.Write(PaxConfig.deviceMap[out_ports[idx]].Name + ", ");
+        } else {
+          Debug.Write(PaxConfig.deviceMap[out_ports[idx]].Name);
+        }
+#endif
+      }
+
+#if DEBUG
+      Debug.WriteLine("");
+#endif
+    }
+  }
+
+  public class PacketProcessor_Chain : PacketProcessor {
+    List<PacketProcessor> chain;
+
+    public PacketProcessor_Chain (List<PacketProcessor> chain) {
+      this.chain = chain;
+    }
+
+    public void packetHandler (object sender, CaptureEventArgs e) {
+      foreach (PacketProcessor pp in chain) {
+        pp.packetHandler (sender, e);
+      }
+    }
+
+    public ForwardingDecision process_packet (int in_port, ref Packet packet)
+    {
+      ForwardingDecision fd = null;
+
+      //We return the ForwardingDecision made by the last element in the chain.
+      foreach (PacketProcessor pp in chain) {
+        fd = pp.process_packet (in_port, ref packet);
+      }
+
+      return fd;
+    }
+  }
+
+#region Aux functions
```

Comment:
Ack, done in https://github.com/niksu/pax/pull/7/commits/3ec4df9aed665a6903097e0166cf3a4e36022509